GEECS-PythonAPI/geecs_python_api/controls/data_acquisition/multiscan_analysis.py

```diff
@@ -0,0 +1,294 @@
+
+        # get grid size
+        grid_row = len(scans.keys())
+        grid_col = len(scans[first_scan]['binned_data'].keys())
+
+        # get global color scale
+        all_pixels = np.concatenate([scans[snum]['binned_data'][bnum]['image'].ravel()
+                                     for snum in list(scans.keys())
+                                     for bnum in list(scans[snum]['binned_data'].keys())
+                                     if scans[snum]['binned_data'][bnum]['image'] is not None])
+        vmin, vmax = 0, all_pixels.max()
+
+        # create figure with appropriate number of subplots
+        fig, ax = plt.subplots(grid_row, grid_col, figsize=(grid_col*3, grid_row*3))
+
+        fig.suptitle(f"MultiScan Visualizer\n{self.scan_parameter}")
+
+        # loop through images
+        for scan_ind, scan_num in enumerate(scans.keys()):
+            scan_item = scans[scan_num]['binned_data']
+            for bin_ind, bin_num in enumerate(scan_item.keys()):
+                bin_item = scan_item[bin_num]
+
+                # display image with adjusted scale
+                ax[scan_ind, bin_ind].imshow(bin_item['image'],
+                                             cmap='plasma', vmin=vmin, vmax=vmax)
+
+                # include scan parameter value as title
+                ax[scan_ind, bin_ind].set_title(f"{bin_item['value']:.2f}")
+
+                # hide axes for cleaner display
+                # ax[scan_ind, bin_ind].axis('off')
+                ax[scan_ind, bin_ind].set_yticklabels([])
+                ax[scan_ind, bin_ind].set_xticklabels([])
+                ax[scan_ind, bin_ind].set_yticks([])
+                ax[scan_ind, bin_ind].set_xticks([])
+
+                # add row labels
+                if bin_ind == 0:
+                    ax[scan_ind, 0].set_ylabel(f"Scan {scan_num}\n{scans[scan_num]['device']}")
+
+                # include reference coordinate (needs modified for each visa screen)
+                if ref_coords:
+                    ax[scan_ind, bin_ind].plot(scans[scan_num]['diode_coords'][0],
+                                               scans[scan_num]['diode_coords'][1],
+                                               color='g', marker='o')
+
+        # for scan_ind, scan_num in enumerate(scans.keys()):
+        #     ax[scan_ind, 0].set_ylabel(f"Scan {scan_num}")
+
+        # plt.tight_layout()
+
+        if save_figure:
+            plt.savefig(save_path, bbox_inches='tight')
+
+        plt.close()
+
+        return
+
+    def run_routine(self, run_analysis=False):
+
+        # run analysis for each scan
+        for key, item in self.scans.items():
+            if run_analysis:
+                self.scans[key] = self.scan_analysis(item)
+            else:
+                self.scans[key] = self.load_analysis_results(item)
+
+        # compile image matrix
+        self.compile_image_matrix(self.scans)
+
+        return
+
```

Comment:
A broad comment for the class as a whole:  you can remove a lot of arguments in class functions by instead assigning them as class variables during the initialization.  IE, you could have a `run_analysis=False` during the `__init__()` then you won't need to pass it in `run_routine()`.  This can potentially simplify a lot of re-reading down the line because all of the options are just in one place instead of scattered throughout the arguments of methods